server/svix-server/src/core/types.rs

```diff
@@ -490,6 +490,42 @@ pub struct ExpiringSigningKey {
     pub expiration: DateTime<Utc>,
 }
 
+#[derive(Clone, Debug, PartialEq, Serialize)]
+pub struct EndpointHeaders(pub HashMap<String, String>);
+json_wrapper!(EndpointHeaders);
+
+impl<'de> Deserialize<'de> for EndpointHeaders {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: serde::Deserializer<'de>,
+    {
+        HashMap::deserialize(deserializer)
+            .map(|x: HashMap<String, String>| {
+                x.into_iter().map(|(k, v)| (k.to_lowercase(), v)).collect()
```

Comment:
Why is moving all the keys to lowercase necessary?

Reply:
This was an attempt to ensure compatibility with existing API, as in my testing it seems to always return lower-case returned results. though in theory it shouldn't matter since header names are case-insensitive per HTTP spec. 

I would suggest leaving it in here for now for the sake of the other code where header names are checked against SENSITIVE header list.

Reply:
Cool, just wanted to make sure it's intentional.